### PR TITLE
Updated corresponding pages due to with Magento Coding Standard release

### PR DESCRIPTION
**Pull request by lenaorobei**
## Purpose of this PR

This PR adds information about [Magento Coding Standard](https://github.com/magento/magento-coding-standard) and fixes outdated pages in this regard.

## Affected URLs

https://devdocs.magento.com/guides/v2.1/coding-standards/code-standard-sniffers.html
https://devdocs.magento.com/guides/v2.2/coding-standards/code-standard-sniffers.html
https://devdocs.magento.com/guides/v2.3/coding-standards/code-standard-sniffers.html
https://devdocs.magento.com/guides/v2.1/coding-standards/code-standard-php.html
https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/templates/template-security.html
https://devdocs.magento.com/guides/v2.1/ext-best-practices/extension-coding/security-performance-data-bp.html
https://devdocs.magento.com/guides/v2.1/ext-best-practices/extension-coding/working-with-arch-bp.html
https://devdocs.magento.com/guides/v2.1/extension-dev-guide/validate/test-module.html
https://devdocs.magento.com/guides/v2.2/extension-dev-guide/validate/test-module.html
https://devdocs.magento.com/guides/v2.3/extension-dev-guide/validate/test-module.html

### Request for Review:
@jeff-matthews 
@dshevtsov 
@paliarush 

whatsnew
Added information about the Magento Coding Standard to the following topics:
- [Code sniffers](https://devdocs.magento.com/guides/v2.3/coding-standards/code-standard-sniffers.html)
- [PHP coding standard](https://devdocs.magento.com/guides/v2.3/coding-standards/code-standard-php.html)
- [Template CSS security](https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/templates/template-security.html)
- [Security, performances, and data handling](https://devdocs.magento.com/guides/v2.3/ext-best-practices/extension-coding/security-performance-data-bp.html)
- [Working with architecture](https://devdocs.magento.com/guides/v2.3/ext-best-practices/extension-coding/working-with-arch-bp.html)
- [Test your component](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/validate/test-module.html)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
@lenaorobei thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/devdocs?pullRequest=4360) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/devdocs?pullRequest=4360) before we can accept your contribution.<br/>

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by lenaorobei** [comment]
Adding more clarity here:
- Moved `code-standard-sniffers.html` content  under `code-standard-php.html` page, because Code Sniffer is a tools that allows to check the code compliance with specific Coding Standard.
- Replaced links to outdated [MEQP Coding Standard](https://github.com/magento/marketplace-eqp) with new one consolidated [Magento Coding Standard](https://github.com/magento/magento-coding-standard).
- Fixed `Templates XSS security` with relevant for 2.3 approach of testing `phtml` templates.

File: guides/v2.1/coding-standards/code-standard-php.md [hunk 1]
```diff
@@ -9,13 +9,30 @@ functional_areas:
   - Standards
 ---
 
-The Magento core development team uses the [PSR-1: Basic Coding Standard](http://www.php-fig.org/psr/psr-1/){:target="_blank"} and [PSR-2: Coding Style Guide](http://www.php-fig.org/psr/psr-2/){:target="_blank"}.
-Magento recommends that developers who create Magento extensions and customizations also use these standards.
+The Magento core development team uses [Magento Coding Standard](https://github.com/magento/magento-coding-standard){:target="_blank"} and it is recommended that developers who create Magento extensions and customizations also use this standard.
 
-Where possible, use [`PHP_CodeSniffer`](https://github.com/squizlabs/PHP_CodeSniffer){:target="_blank"} to automatically enforce these standards.
-Otherwise, you must apply these standards and requirements through rigorous code review.
+Magento Coding Standard provides the set of rules which cover following:
```
Comment (jeff-matthews) [inline guides/v2.1/coding-standards/code-standard-php.md]:
```suggestion
The Magento Coding Standard provides a set of rules that covers the following:
```

File: guides/v2.1/coding-standards/code-standard-php.md [hunk 2]
```diff
@@ -9,13 +9,30 @@ functional_areas:
   - Standards
 ---
 
-The Magento core development team uses the [PSR-1: Basic Coding Standard](http://www.php-fig.org/psr/psr-1/){:target="_blank"} and [PSR-2: Coding Style Guide](http://www.php-fig.org/psr/psr-2/){:target="_blank"}.
-Magento recommends that developers who create Magento extensions and customizations also use these standards.
+The Magento core development team uses [Magento Coding Standard](https://github.com/magento/magento-coding-standard){:target="_blank"} and it is recommended that developers who create Magento extensions and customizations also use this standard.
```
Comment (jeff-matthews) [inline guides/v2.1/coding-standards/code-standard-php.md]:
```suggestion
The Magento core development team uses the [Magento Coding Standard](https://github.com/magento/magento-coding-standard){:target="_blank"}. We recommend that developers who create Magento extensions and customizations also use this standard.
```

File: guides/v2.1/coding-standards/code-standard-php.md [hunk 3]
```diff
@@ -9,13 +9,30 @@ functional_areas:
...
 
-Where possible, use [`PHP_CodeSniffer`](https://github.com/squizlabs/PHP_CodeSniffer){:target="_blank"} to automatically enforce these standards.
-Otherwise, you must apply these standards and requirements through rigorous code review.
+Magento Coding Standard provides the set of rules which cover following:
+* [PSR-1]{:target="_blank"} and [PSR-2]{:target="_blank"} compliance
+* the use of insecure functions
```
Comment (jeff-matthews) [inline guides/v2.1/coding-standards/code-standard-php.md]:
```suggestion
* The use of insecure functions
```

File: guides/v2.1/coding-standards/code-standard-php.md [hunk 4]
```diff
@@ -9,13 +9,30 @@ functional_areas:
...
+* unescaped output
```
Comment (jeff-matthews) [inline guides/v2.1/coding-standards/code-standard-php.md]:
```suggestion
* Unescaped output
```

File: guides/v2.1/coding-standards/code-standard-php.md [hunk 5]
```diff
@@ -9,13 +9,30 @@ functional_areas:
...
+* the use of deprecated PHP functions
```
Comment (jeff-matthews) [inline guides/v2.1/coding-standards/code-standard-php.md]:
```suggestion
* The use of deprecated PHP functions
```

File: guides/v2.1/coding-standards/code-standard-php.md [hunk 6]
```diff
@@ -9,13 +9,30 @@ functional_areas:
...
+* PHP code syntax
+* naming convention
```
Comment (jeff-matthews) [inline guides/v2.1/coding-standards/code-standard-php.md]:
```suggestion
* Naming convention
```

File: guides/v2.1/coding-standards/code-standard-php.md [hunk 7]
```diff
@@ -9,13 +9,30 @@ functional_areas:
...
+* the use of PHP superglobals
```
Comment (jeff-matthews) [inline guides/v2.1/coding-standards/code-standard-php.md]:
```suggestion
* The use of PHP superglobals
```

File: guides/v2.1/coding-standards/code-standard-php.md [hunk 8]
```diff
@@ -9,13 +9,30 @@ functional_areas:
...
+* empty code blocks
```
Comment (jeff-matthews) [inline guides/v2.1/coding-standards/code-standard-php.md]:
```suggestion
* Empty code blocks
```

File: guides/v2.1/coding-standards/code-standard-php.md [hunk 9]
```diff
@@ -9,13 +9,30 @@ functional_areas:
...
+* improper exception handling
```
Comment (jeff-matthews) [inline guides/v2.1/coding-standards/code-standard-php.md]:
```suggestion
* Improper exception handling
```

File: guides/v2.1/coding-standards/code-standard-php.md [hunk 10]
```diff
@@ -9,13 +9,30 @@ functional_areas:
...
+* raw SQL queries and many other general PHP and Magento specific code issues.
```
Comment (jeff-matthews) [inline guides/v2.1/coding-standards/code-standard-php.md]:
```suggestion
* Raw SQL queries and many other general PHP and Magento-specific code issues.
```

File: guides/v2.1/coding-standards/code-standard-php.md [hunk 11]
```diff
@@ -9,13 +9,30 @@ functional_areas:
...
 
-### Class name resolution standard
+### Coding Standard Compliance
+
+Developers should consistently use [PHP_CodeSniffer]{:target="_blank"} to enhance the readability of the code and ensure that it meets Magento Coding Standard. [PHP_CodeSniffer]{:target="_blank"} is the most popular tool in use throughout the {% glossarytooltip bf703ab1-ca4b-48f9-b2b7-16a81fd46e02 %}PHP{% endglossarytooltip %} development community.
```
Comment (jeff-matthews) [inline guides/v2.1/coding-standards/code-standard-php.md]:
```suggestion
Developers should consistently use [PHP_CodeSniffer]{:target="_blank"} to enhance the readability of the code and ensure that it meets the Magento Coding Standard. [PHP_CodeSniffer]{:target="_blank"} is the most popular tool in use throughout the {% glossarytooltip bf703ab1-ca4b-48f9-b2b7-16a81fd46e02 %}PHP{% endglossarytooltip %} development community.
```

File: guides/v2.1/coding-standards/code-standard-php.md [hunk 12]
```diff
@@ -9,13 +9,30 @@ functional_areas:
...
+* the use of deprecated PHP functions
+* PHP code syntax
+* naming convention
+* the use of PHP superglobals
+* empty code blocks
+* improper exception handling
+* raw SQL queries and many other general PHP and Magento specific code issues.
 
-### Class name resolution standard
+### Coding Standard Compliance
```
Comment (jeff-matthews) [inline guides/v2.1/coding-standards/code-standard-php.md]:
```suggestion
### Coding standard compliance
```

File: guides/v2.1/ext-best-practices/extension-coding/security-performance-data-bp.md [hunk 13]
```diff
@@ -13,7 +13,7 @@ You should make sure that your {% glossarytooltip 55774db9-bf9d-40f3-83db-b10cc5
...
 
-  For a list of discouraged low-level functions, we suggest you look at the [list of discouraged functions](https://github.com/magento/marketplace-eqp/blob/master/MEQP1/Sniffs/Security/DiscouragedFunctionSniff.php){:target="_blank"} for [Magento Extension Quality Program Coding Standard](https://github.com/magento/marketplace-eqp){:target="_blank"}.
+  For a list of discouraged low-level functions, we suggest you look at the [list of discouraged functions](https://github.com/magento/magento-coding-standard/blob/develop/Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php){:target="_blank"} for [Magento Coding Standard](https://github.com/magento/magento-coding-standard){:target="_blank"}.
```
Comment (jeff-matthews) [inline guides/v2.1/ext-best-practices/extension-coding/security-performance-data-bp.md]:
```suggestion
For a list of discouraged low-level functions, review the [`Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php`](https://github.com/magento/magento-coding-standard/blob/develop/Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php){:target="_blank"} file and the [Magento Coding Standard](https://github.com/magento/magento-coding-standard){:target="_blank"}.
```

File: guides/v2.2/extension-dev-guide/validate/test-module.md [hunk 14]
```diff
@@ -46,7 +46,7 @@ Before you publish your component, test installing it using the [Magento Compone
 
 See these resources for testing in {% glossarytooltip bf703ab1-ca4b-48f9-b2b7-16a81fd46e02 %}PHP{% endglossarytooltip %} and validating Magento components:
 
-* [Magento Marketplace Extension Quality Program Coding Standard] provides a set of rules and sniffs for [PHP_CodeSniffer] tool
+* [Magento Coding Standard] provides a set of rules and sniffs for [PHP_CodeSniffer] tool
```
Comment (jeff-matthews) [inline guides/v2.2/extension-dev-guide/validate/test-module.md]:
```suggestion
* The [Magento Coding Standard] provides a set of rules and sniffs for the [PHP_CodeSniffer] tool
```

File: guides/v2.1/extension-dev-guide/validate/test-module.md [hunk 15]
```diff
@@ -47,7 +47,7 @@ Before you publish your component, test installing it using the [Magento Compone
 
 See these resources on testing in {% glossarytooltip bf703ab1-ca4b-48f9-b2b7-16a81fd46e02 %}PHP{% endglossarytooltip %} and validating Magento components:
 
-* [Magento Marketplace Extension Quality Program Coding Standard] provides a set of rules and sniffs for [PHP_CodeSniffer] tool
+* [Magento Coding Standard] provides a set of rules and sniffs for [PHP_CodeSniffer] tool
```
Comment (jeff-matthews) [inline guides/v2.1/extension-dev-guide/validate/test-module.md]:
```suggestion
* The [Magento Coding Standard] provides a set of rules and sniffs for the [PHP_CodeSniffer] tool
```

File: guides/v2.3/extension-dev-guide/validate/test-module.md [hunk 16]
```diff
@@ -44,7 +44,7 @@ Before you publish your component, test installing it using the [Magento Compone
 
 See these resources for testing in {% glossarytooltip bf703ab1-ca4b-48f9-b2b7-16a81fd46e02 %}PHP{% endglossarytooltip %} and validating Magento components:
 
-* [Magento Marketplace Extension Quality Program Coding Standard] provides a set of rules and sniffs for [PHP_CodeSniffer] tool
+* [Magento Coding Standard] provides a set of rules and sniffs for [PHP_CodeSniffer] tool
```
Comment (jeff-matthews) [inline guides/v2.3/extension-dev-guide/validate/test-module.md]:
```suggestion
* The [Magento Coding Standard] provides a set of rules and sniffs for the [PHP_CodeSniffer] tool
```

File: guides/v2.1/coding-standards/code-standard-php.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (lenaorobei) [reply guides/v2.1/coding-standards/code-standard-php.md]: I don't see an issue here. It looks good to me that we explicitly declare PHP standard we are following.

**Comment by lenaorobei** [comment]
@jeff-matthews thank you fo the review and suggestions.

**Comment by jeff-matthews** [comment]
running tests

**Comment by jeff-matthews** [comment]
Tests failed. Here's the log.

+ rake test:cicd
[35mChecking links with htmlproofer...[0m
Running ["LinkChecker::DoubleSlashCheck", "ImageCheck", "ScriptCheck", "LinkCheck"] on ["_site"] on *.html... 


Ran on 4098 files!

```
- internally linking to /guides/v2.1/coding-standards/code-standard-sniffers.html, which does not exist
  *  _site/guides/v2.1/coding-standards/bk-coding-standards.html (line 208)
     <a href="/guides/v2.1/coding-standards/code-standard-sniffers.html">PHP code sniffers</a>
- internally linking to /guides/v2.2/coding-standards/code-standard-sniffers.html, which does not exist
  *  _site/guides/v2.2/coding-standards/bk-coding-standards.html (line 208)
     <a href="/guides/v2.2/coding-standards/code-standard-sniffers.html">PHP code sniffers</a>
- internally linking to /guides/v2.3/coding-standards/code-standard-sniffers.html, which does not exist
  *  _site/guides/v2.3/coding-standards/bk-coding-standards.html (line 208)
     <a href="/guides/v2.3/coding-standards/code-standard-sniffers.html">PHP code sniffers</a>
rake aborted!
HTML-Proofer found 3 failures!```

**Comment by jeff-matthews** [comment]
@lenaorobei, please remove the link to the deleted Code Sniffer file in the following file:

> guides/v2.1/coding-standards/bk-coding-standards.md

**Comment by lenaorobei** [comment]
@jeff-matthews, done.

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**